prov/efa/src/efa_user_info.c

```diff
@@ -332,7 +332,82 @@ int efa_user_info_alter_rdm(int version, struct fi_info *info, const struct fi_i
 }
 
 /**
- * @brief get a list of rdm info the fit user's requirements
+ * @brief update EFA direct info to match user hints
+ *
+ * the input info is a duplicate of prov info, which matches
+ * the capability of the EFA device. This function tailor it
+ * so it matches user provided hints
+ *
+ * @param	version[in]	libfabric API version
+ * @param	info[in,out]	info to be updated
+ * @param	hints[in]	user provided hints
+ * @return	0 on success
+ * 		negative libfabric error code on failure
+ */
+static
+int efa_user_info_alter_direct(int version, struct fi_info *info, const struct fi_info *hints)
+{
+	if (hints && (hints->caps & FI_HMEM)) {
+		/*
+		 * FI_HMEM is a primary capability, therefore only check
+		 * (and cliam) its support when user explicitly requested it.
+		 */
+		if (!efa_user_info_should_support_hmem(version)) {
```

Comment:
I don't think we need to call this function, as it has some API version check & device rdma check which is only useful for efa-rdm

Reply:
Discussed offline. Will keep `ofi_hmem_p2p_disabled` check but move it to `efa_prov_info`